### PR TITLE
NEOS-782:support for primary key column and badges in the table

### PR DESCRIPTION
**Pull request by evisdrenova**

### CODE REVIEW

**Review by nickzelei — COMMENTED** [review]
would be great to see a screenshot of what this looks like in the table

File: frontend/apps/web/components/jobs/SchemaTable/SchemaColumns.tsx [hunk 1]
```diff
@@ -12,19 +20,27 @@ import {
...
 import { HTMLProps, useEffect, useRef } from 'react';
 import { SchemaColumnHeader } from './SchemaColumnHeader';
-import { Row } from './SchemaPageTable';
+import { Row as RowData } from './SchemaPageTable';
 import TransformerSelect from './TransformerSelect';
 
 interface Props {
   transformers: Transformer[];
+  primaryConstraints?: { [key: string]: PrimaryConstraint };
+  foreignConstraints?: { [key: string]: ForeignConstraintTables };
```
[the remark this answers is not in the document]
Comment (evisdrenova) [reply frontend/apps/web/components/jobs/SchemaTable/SchemaColumns.tsx:39]: updated

File: frontend/apps/web/components/jobs/SchemaTable/SchemaColumns.tsx [hunk 2]
```diff
@@ -12,19 +20,27 @@ import {
...
 }
 
-export function getSchemaColumns(props: Props): ColumnDef<Row>[] {
-  const { transformers } = props;
+export function getSchemaColumns(props: Props): ColumnDef<RowData>[] {
+  const { transformers, primaryConstraints } = props;
```
[the remark this answers is not in the document]
Comment (evisdrenova) [reply frontend/apps/web/components/jobs/SchemaTable/SchemaColumns.tsx:43]: yup holdovers, removed

**Comment by evisdrenova** [comment]
> would be great to see a screenshot of what this looks like in the table

Here's a loom:
https://www.loom.com/share/a4104bf6057b4bad9b14630c02297943

**Comment by nickzelei** [comment]
> > would be great to see a screenshot of what this looks like in the table
> 
> Here's a loom: https://www.loom.com/share/a4104bf6057b4bad9b14630c02297943

Nice. A small thing that could make help is to color the badges to make them stick out a little more.
A separate improvement could be to allow searching in the constraint box for those constraints. Or maybe that is possible to do right now, now sure.

But if you have something like:

 public.users.id

public.user_accounts.user_id -- foreign keyed to public.users.id

if in the constraints box you searched for "users.id", if those two records showed up. Seems doable since you are surfacing that in the highlighted text. could add a fuzzy search on it for the constraints column.

**Review by nickzelei — APPROVED** [review]

**Pull request merged**